File: app/src/main/java/com/zlm/kt/data/DataManager.java
package com.zlm.kt.data;

import android.content.Context;

import com.zlm.kt.connect.ApiClientImp;
import com.zlm.kt.connect.ApiService;

import javax.inject.Inject;
import javax.inject.Singleton;

/**
 * @author Milla
 * @create 2020/3/29
 */

@Singleton
public class DataManager implements DataManagerImp {

    private final Context context;
    private final ApiClientImp apiClient;

    // -------------------------------------------
    @Inject
    public DataManager(Context context, ApiClientImp apiClient) {
        this.context = context;
        this.apiClient = apiClient;
    }

    // -------------------------------------------
    @Override
    public ApiService getApi() {
        return apiClient.getApi();
    }

    // -------------------------------------------
    @Override
    public Context getContext() {
        return context;
    }

    // -------------------------------------------
}

File: app/src/main/java/com/zlm/kt/data/DataManager.kt
package com.zlm.kt.data

import android.content.Context
import com.zlm.kt.connect.ApiClientImp
import com.zlm.kt.connect.ApiService
import javax.inject.Inject
import javax.inject.Singleton

/**
 * @author Milla
 * @create 2020/3/29
 */
@Singleton
class DataManager
    @Inject
    constructor(
        override val context: Context, private val apiClient: ApiClientImp) : DataManagerImp {

    override val api: ApiService
        get() = apiClient.api

}